lib/praxis/application.rb

```diff
@@ -51,28 +51,30 @@ def initialize
     
 
     def setup(root: '.')
-      @root = Pathname.new(root).expand_path
-
-      builtin_handlers = {
-          'json'                  => Praxis::Handlers::JSON,
-          'x-www-form-urlencoded' => Praxis::Handlers::WWWForm
-      }
-      # Register built-in handlers unless the app already provided its own
-      builtin_handlers.each_pair do |name, handler|
-        self.handler(name, handler) unless handlers.key?(name)
-      end
-
-      @bootloader.setup!
-
-      @builder.run(@router)
-      @app = @builder.to_app
-
-      Notifications.subscribe 'rack.request.all'.freeze do |name, start, finish, _id, payload|
-        duration = (finish - start) * 1000
-        Stats.timing(name, duration)
-
-        status, _, _ = payload[:response]
-        Stats.increment "rack.request.#{status}"
+      if @app.nil?
```

Comment:
it's probably a pedantic thing but, I think it'd be more clear (both for changed lines noise review, and code itself) to do the change by simply adding a `return unless @app.nil?`

no?